ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-bgp-vrfs.j2

```diff
@@ -56,7 +56,10 @@
                                                          tenant.enable_mlag_ibgp_peering_vrfs,
                                                          true) %}
 {%                         if configure_mlag_ibgp_peering %}
-        {{ switch.mlag_peer_l3_ip | arista.avd.default(switch.mlag_peer_ip) }}:
+{%                             if vrf.mlag_ibgp_peering_ipv4_pool is arista.avd.defined %}
+{%                                 set mlag_ibgp_peering_peer_ip = vrf.mlag_ibgp_peering_ipv4_pool | ansible.netcommon.ipaddr('subnet') | ansible.netcommon.ipmath(switch.mlag_peer_id - 1) %}
```

Comment:
Correct. Will change the code.

Reply:
But the only problem that I see was with the user definition. For example: switch pairs with ids (1,3) and (2,4). The switch 3 and switch 2 will end up having same IP.

I still agree with your suggestion as taking id and peer_id to calculate the IPs will not keep the p2p links in the same subnet (/31). For example (1,3) -> .0/31 and .2/31 are not in the same subnet.

Another problem with non-consecutive switch ids is: **The ip and peer_ip** might end up in a different /31 subnets. For example the switch pair (2,4) -> .1/31 and .2/31 (different subnets)

Reply:
Probably I need a stronger algorithm to solve all these issues.

Reply:
@tgodaA:

if I have two pairs of ids: A = (2,4) and B = (3,5)

it would lead to 
A:
1. id = 2
2. primary_ip_suffix = 2 - 2%2 = 2
3. secondary_ip_suffix = primary_ip_suffix + 1 = 3


B: 
1. id = 3
2. primary_ip_suffix = 3 - 3%2 = 2
3. secondary_ip_suffix = primary_ip_suffix + 1 = 3

so there would be a clash